phoenix-core/src/main/java/org/apache/phoenix/iterate/RegionContext.java

```diff
@@ -0,0 +1,15 @@
+package org.apache.phoenix.iterate;
+
+import org.apache.hadoop.hbase.coprocessor.RegionCoprocessorEnvironment;
+
+/**
+ * Maintains coprocessor environment state to
+ * make region observer functionality work for
+ * non-coprocessor environment
+ */
+public interface RegionContext {
+
+  /** @return the Coprocessor environment */
+  RegionCoprocessorEnvironment getEnvironment();
```

Comment:
If this is the only method, we don't really need this interface. Why not just keep using RegionCoprocessorEnvironment in our APIs and you can just instantiate you snapshot-based one prior to getting the RegionScanner?